backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -135,46 +137,46 @@ public void updateUser(String oldUsername, IdentityAttributes userIdentity) {
         user.getProfile().setEmail(userIdentity.getUsername());
         user.getProfile().setLogin(userIdentity.getUsername());
         user.update();
+
+        return getOrganizationRoleClaimsForUser(user);
     }
 
-    public OrganizationRole updateUserRole(String username, OrganizationRole role) {
+    public OrganizationRole updateUserRole(String username, Organization org, OrganizationRole role) {
         UserList users = _client.listUsers(username, null, null, null, null);
         if (!users.iterator().hasNext()) {
             throw new IllegalGraphqlArgumentException("Cannot update role of Okta user with unrecognized username");
         }
         User user = users.single();
 
-        Set<String> orgExternalIds = new HashSet<>();
+        String orgId = org.getExternalId();
 
         // Remove user from old groups
         for (Group g : user.listGroups()) {
```

Comment:
This nested loop was kind of confusing to read, though obviously the diffs didn't help. I think it is equivalent to the following, which I think is easier to follow:

    for (Group g : user.listGroups()) {
        if (g.getType() != GroupType.OKTA_GROUP) {
           continue;
        }
        String groupName = g.getProfile().getName();
        for (OrganizationRole r : OrganizationRole.values()) {
            if (r == OrganizationRole.USER) {
                continue;
            }
            if (generateGroupName(orgId, role).equals(groupName)) {
                g.removeUser(user.getId());
                break;
            }
        }
    }

If I'm right about that equivalence, then I think there's a further rearrangement that might be more efficient and possibly more readable:

    Set<String> roleGroups = Stream.of(OrganizationRole.values()).filter(r -> r != OrganizationRole.getDefault() && r != role)
            .map(r->generateGroupName(orgId, r))
            .collect(Collectors.toSet());
    for (Group g : user.listGroups()) {
        if (g.getType() == GroupType.OKTA_GROUP && roleGroups.contains(g.getProfile().getName()) {
           g.removeUser(user.getId());
        }
    }

Reply:
good catch :)